lib/chef/knife/winrm_knife_base.rb

```diff
@@ -99,71 +109,68 @@ def create_winrm_session(options={})
+              resolve_winrm_transport == :kerberos ||
+              user.include?("\\") ||
+              user.include?("@")
+              user
+            else
+              ".\\#{user}"
+            end
           end
 
-          def resolve_winrm_kerberos_options
-            if config.keys.any? {|k| k.to_s =~ /kerberos/ }
-              @session_opts[:transport] = :kerberos
-              @session_opts[:keytab] = locate_config_value(:kerberos_keytab_file) if locate_config_value(:kerberos_keytab_file)
-              @session_opts[:realm] = locate_config_value(:kerberos_realm) if locate_config_value(:kerberos_realm)
-              @session_opts[:service] = locate_config_value(:kerberos_service) if locate_config_value(:kerberos_service)
-            end
+          def resolve_winrm_session_timeout
+            locate_config_value(:session_timeout).to_i * 60 if locate_config_value(:session_timeout)
           end
 
-          def resolve_winrm_transport_options
-            @session_opts[:disable_sspi] = true
-            @session_opts[:transport] = locate_config_value(:winrm_transport).to_sym unless @session_opts[:transport] == :kerberos
-            if negotiate_auth? && @session_opts[:transport] == :ssl
-                Chef::Log.debug("Trying WinRM communication with negotiate authentication and :ssl transport")
-            elsif use_windows_native_auth?
-              load_windows_specific_gems
-              @session_opts[:transport] = :sspinegotiate
-              @session_opts[:disable_sspi] = false
-            end
+          def resolve_winrm_basic_auth
+            locate_config_value(:winrm_authentication_protocol) == "basic"
           end
 
-          def resolve_winrm_ssl_options
-            @session_opts[:ca_trust_path] = locate_config_value(:ca_trust_file) if locate_config_value(:ca_trust_file)
-            @session_opts[:no_ssl_peer_verification] = no_ssl_peer_verification?(@session_opts[:ca_trust_path])
-            warn_no_ssl_peer_verification if @session_opts[:no_ssl_peer_verification]
+          def resolve_winrm_kerberos_options
+            kerberos_opts = {}
+            kerberos_opts[:keytab] = locate_config_value(:kerberos_keytab_file) if locate_config_value(:kerberos_keytab_file)
+            kerberos_opts[:realm] = locate_config_value(:kerberos_realm) if locate_config_value(:kerberos_realm)
+            kerberos_opts[:service] = locate_config_value(:kerberos_service) if locate_config_value(:kerberos_service)
+            kerberos_opts
           end
 
-          def resolve_winrm_auth_settings
-            winrm_auth_protocol = locate_config_value(:winrm_authentication_protocol)
-            if ! Chef::Knife::WinrmBase::WINRM_AUTH_PROTOCOL_LIST.include?(winrm_auth_protocol)
-              ui.error "Invalid value '#{winrm_auth_protocol}' for --winrm-authentication-protocol option."
-              ui.info "Valid values are #{Chef::Knife::WinrmBase::WINRM_AUTH_PROTOCOL_LIST.join(",")}."
-              exit 1
```

Comment:
No I moved it to `validate_options!` which is the method responsible for warning and exiting on bad options.